client/my-sites/plugins/use-preinstalled-premium-plugin/index.js

```diff
@@ -25,9 +25,10 @@ export default function usePreinstalledPremiumPlugin( pluginSlug ) {
 	);
 
 	const isPreinstalledPremiumPluginActive = useSelector( ( state ) => {
-		if ( ! preinstalledPremiumPlugin || ! selectedSiteId ) {
-			return false;
+		if ( preinstalledPremiumPlugin && selectedSiteId ) {
```

Comment:
How does this work? If the plugin is a pre-installed one, and the user is using calypso with a site selected this is going to return true. 

If I have a jetpack site, without the jetpack search plugin and I'm using calypso to look at the jetpack search plugin, this is going to return true isn't it?

Reply:
Jetpack the plugin comes bundled with Jetpack Search, as a module.

Reply:
Doh, of course. And even on self-hosted, you'll always need JP to access calypso anyway.

What happens if you have say Jetpack Protect, or Jetpack Boost installed? Presumably they won't come with Jetpack Search module, but will have a JP connection.

Reply:
I don't think it matters? The list is hardcoded in `PREINSTALLED_PREMIUM_PLUGINS` constant, and only contains Jetpack Search. 

From this logic PoV it's not really important what the site actually contains. It just checks if the slug is present in the hardcoded constant. If not, it continues with the existing logic (checking if the plugin is actually on the target site).

Reply:
> From this logic PoV it's not really important what the site actually contains

Either this function is really confusingly named, or that's not true. 

If this function `isPreinstalledPremiumPluginActive` is being called for a plugin that isn't in the hard-coded list it should be able to return false immediately, as it's not in the hard-coded list. With these changes that doesn't happen.

To go through the conditions:

 1. preinstalledPremiumPlugin will be false, so it won't check selectedSiteId and won't enter the first guard clause, which returns true
 2. It will get to the second clause "Always active on simple sites" and if it's a simple site, it will enter the guard clause and return true, even if it's not a preinstalled premium plugin. If it's an atomic site, it will continue
 3. Third clause "always active on atomic sites that don't have the plugin installed". It's an atomic site, so if the plugin is not installed it will enter the guard clause and return true
 4. It will return true if the plugin is active on the site



### Jetpack sites
It doesn't matter right now, but previously it would work for JP sites too.


Given a site with Jetpack Boost installed but not the full Jetpack plugin.

preinstalledPremiumPlugin is true (looking for Jetpack Search) and selectedSiteId is true (have a domain in the url that Calypso understands)

## Before: 

1. preInstalledPremiumPlugin is true and selectedSiteId is true, so the first guard clause doesn't apply
2. The site is not AT but is JP so the second guard clause doesn't apply
3. The site is not AT so the third guard close doesn't apply
4. The site return true/false depending on whether Jetpack Search is active

## After:

1. preInstalledPremiumPlugin is true and selectedSiteId is true, so the first guard clause applies and the method returns true, despite the fact we never checked if it is installed on the site.

Reply:
I think the naming is a bit confusing, because Jetpack Search is a module but also a standalone plugin.  But, yeah, you're right. So it's not always preinstalled. 

> It doesn't matter right now, but previously it would work for JP sites too.

Actually, the Jetpack Search product is available in the Marketplace screen and can be purchased for JP sites too from that section.